spec/services/search/classified_listing_spec.rb

```diff
@@ -100,4 +100,159 @@
       described_class.delete_index(index_name: other_name)
     end
   end
+
+  describe "::search_documents" do
+    let(:classified_listing1) { create(:classified_listing) }
+    let(:classified_listing2) { create(:classified_listing) }
+    let(:classified_listing3) { create(:classified_listing) }
+    let(:classified_listing4) { create(:classified_listing) }
+    let(:classified_listing5) { create(:classified_listing) }
+    let(:classified_listings) { [classified_listing1, classified_listing2, classified_listing3, classified_listing4, classified_listing5] }
+
+    def index_documents(resources)
+      resources.each(&:index_to_elasticsearch_inline)
+      described_class.refresh_index
+    end
+
+    it "parses classified_listing document hits from search response" do
+      mock_search_response = { "hits" => { "hits" => {} } }
+      allow(described_class).to receive(:search) { mock_search_response }
+      described_class.search_documents(params: {})
+      expect(described_class).to have_received(:search).with(body: a_kind_of(Hash))
+    end
+
+    context "with a query" do
+      # classified_listing_search is a copy_to field including:
+      # body_markdown, location, slug, tags, and title
+      it "searches by classified_listing_search" do
+        allow(classified_listing1).to receive(:body_markdown).and_return("body_markdown with test")
```

Comment:
Since most of these are ClassifiedListing attributes you could just update them on the listing's themselves. I was stubbing things in my specs bc they were being pulled from related models and were not chat channel membership attributes themselves. 

Same for all the stubbing below